src/gui/mainwindow.cpp

```diff
@@ -452,18 +453,21 @@ MainWindow::MainWindow(QWidget *parent)
 
     qDebug("GUI Built");
 #ifdef Q_OS_WIN
-    if (!pref->neverCheckFileAssoc() && (!Preferences::isTorrentFileAssocSet() || !Preferences::isMagnetLinkAssocSet())) {
-        if (QMessageBox::question(this, tr("Torrent file association"),
-                                  tr("qBittorrent is not the default application to open torrent files or Magnet links.\nDo you want to associate qBittorrent to torrent files and Magnet links?"),
-                                  QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes) == QMessageBox::Yes) {
-            Preferences::setTorrentFileAssoc(true);
-            Preferences::setMagnetLinkAssoc(true);
-        }
-        else {
-            pref->setNeverCheckFileAssoc();
-        }
-    }
-#endif
+    if (!pref->neverCheckFileAssoc() && !Utils::OS::isAssociationsSet()) {
+        const QString dlgTitle {tr("Torrent file association")};
+        const QString dlgText {((QSysInfo::WindowsVersion <= QSysInfo::WV_WINDOWS7)
+                    ? tr("qBittorrent is not the default application to open torrent files or magnet links.\n"
+                         "Do you want to associate qBittorrent with torrent files and magnet links?")
+                    : tr("Do you want to set qBittorrent as the default program for torrent files and magnet links?\n"
+                         "You can do it later using \"Default Programs\" dialog from the Control Panel."))};
+
+        int dialogResult = QMessageBox::question(this, dlgTitle, dlgText, QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
```

Comment:
use `QMessageBox::StandardButton` instead of int?

Reply:
This method returns `int`.

Reply:
It can also be `QMessageBox::StandardButton`
http://doc.qt.io/qt-5/qmessagebox.html#question